keras/layers/variational.py

```diff
@@ -0,0 +1,105 @@
+from keras import backend as K
+from keras.layers.core import Layer
+from keras import initializations, activations
+
+from keras.regularizers import GaussianKL
+
+
+class VariationalDense(Layer):
+    """VariationalDense
+    Hidden layer for Variational Autoencoding Bayes method - Kigma and
+    Welling 2013.
+    This layer projects the input twice to calculate the mean and variance
+    of a Gaussian distribution. During training, the output is sampled from
+    that distribution as mean + random_noise * variance, during testing the
+    output is the mean, i.e the expected value of the encoded distribution.
```

Comment:
I believe this docstring is difficult to follow for those not already familiar with variational autoencoders. Maybe it would be clearer to add a couple lines to describe mathematically how the mean and variance are obtained, and to specify what are the assumptions made about the data?